src/Main.ts

```diff
@@ -1120,6 +1120,74 @@ export class Main {
         }
     }
 
+    public async getChannelInfo(opts: {
+        slack_channel_id: string,
+        slack_bot_token: string,
+        team_id: string,
+    }): Promise<ConversationsInfoResponse|void> {
+        let slackClient: WebClient | undefined;
+        let teamEntry: TeamEntry | null = null;
+        let teamId: string = opts.team_id;
+
+        const existingChannel = opts.slack_channel_id ? this.rooms.getBySlackChannelId(opts.slack_channel_id) : null;
+
+        if (existingChannel) {
+            throw Error("Channel is already bridged! Unbridge the channel first.");
+        }
+
+        if (opts.slack_bot_token) {
+            if (!opts.slack_bot_token.startsWith("xoxb-")) {
+                throw Error("Provided token is not a bot token. Ensure the token starts with xoxb-");
+            }
+            // We may have this team already and want to update the token, or this might be new.
+            // But first check that the token works.
+            try {
+                teamId = await this.clientFactory.upsertTeamByToken(opts.slack_bot_token);
+                log.info(`Found ${teamId} for token`);
+            } catch (ex) {
+                log.error("Failed to action link because the token couldn't used:", ex);
+                throw Error("Token did not work, unable to get team");
+            }
+        }
+
+        // else, assume we have a teamId
+        if (teamId) {
+            try {
+                slackClient = await this.clientFactory.getTeamClient(teamId);
+            } catch (ex) {
+                log.error("Failed to action link because the team client couldn't be fetched:", ex);
+                throw Error("Team is known, but unable to get team client");
+            }
+
+            teamEntry = await this.datastore.getTeam(teamId);
+            if (!teamEntry) {
+                throw Error("Team ID provided, but no team found in database");
+            }
+        }
+
+        let channelInfo: ConversationsInfoResponse | undefined;
+        if (slackClient && opts.slack_channel_id && opts.team_id) {
```

Comment:
As per above, we can assume a teamId and a channelId

Reply:
Removed the two conditions.